openqaoa/problems/binpacking.py

```diff
@@ -0,0 +1,324 @@
+            if self.solution[var] is None:
+                vars_[var] = mdl.binary_var(var)
+            else:
+                vars_[var] = self.solution[var]
+        objective = mdl.sum([vars_[y] for y in vars_.keys() if y[0] == "y"])
+        self.vars_pos = {var.name:n for n, var in enumerate(mdl.iter_binary_vars())}
+
+        mdl.minimize(objective)
+        if self.simplifications:
+            list_items = range(1, self.n_items)
+        else:
+            list_items = range(self.n_items)
+        for i in list_items:
+            # First set of constraints: the items must be in any bin
+            self.eq_constraints[f"eq_{i}"] = [[self.vars_pos[f"x_{i}_{j}"] for j in range(self.n_bins)], [1]]
+            mdl.add_constraint(mdl.sum(vars_[f"x_{i}_{j}"] for j in range(self.n_bins)) == 1)
+        if self.include_ineqs:
+            for j in range(self.n_bins):
+                # Second set of constraints: weight constraints
+                mdl.add_constraint(
+                    mdl.sum((self.weights[i]/self.weight_capacity) * vars_[f"x_{i}_{j}"] for i in range(self.n_items)) <=  vars_[f"y_{j}"]
+                )
+                if self.simplifications and j < self.min_bins:
+                    if j == 0:
+                        self.ineq_constraints[f"ineq_{j}"] = [[self.vars_pos[f"x_{i}_{j}"] for i in list_items], [self.weight_capacity - self.weights[0]]]
+                    else:
+                        self.ineq_constraints[f"ineq_{j}"] = [[self.vars_pos[f"x_{i}_{j}"] for i in list_items], [self.weight_capacity]]
+                        
+                else:
+                    self.ineq_constraints[f"ineq_{j}"] = [[self.vars_pos[f"x_{i}_{j}"] for i in list_items], [self.vars_pos[f"y_{j}"]]]
+
+        return mdl
+
+    def get_qubo_problem(self):
+        """
+        Returns the QUBO encoding of this problem.
+        
+        Returns
+        -------
+            The QUBO encoding of this problem.
+        """
+        if self.method == "slack":
+            qubo = FromDocplex2IsingModel(self.cplex_model, multipliers=self.penalty).ising_model
+        elif self.method == "unbalanced":
+            qubo = FromDocplex2IsingModel(self.cplex_model, multipliers=self.penalty[0],
+                                          unbalanced_const=True, strength_ineq=self.penalty[1:]).ising_model
+        return qubo
+    
+    def classical_solution(self, string=False):
+        docplex_sol = self.cplex_model.solve()
+        if string:
+            solution = ""
+        else:
+            solution = self.solution.copy()
+        for var in self.cplex_model.iter_binary_vars():
+            if string:
+                solution += str(int(np.round(docplex_sol.get_value(var), 1)))
+            else:
+                solution[var.name] = int(np.round(docplex_sol.get_value(var), 1))
+        if not docplex_sol.is_valid_solution():
+            raise ValueError("The Cplex solver does not find a solution.")
+        return solution
```

Comment:
This function looks generalizable for any cplex_model object. As mentioned, maybe its better for this method to be a function in the utilities class where we do something like `solve_with_cplex`